src/algorithm/spantree.rs

```diff
@@ -0,0 +1,233 @@
+use crate::data_structure::{
+    graph_dcel::GraphDCEL,
+    link_graph::{LinkDart, LinkFace, LinkGraphIter, LinkVertex},
+};
+use std::collections::{HashMap, HashSet, VecDeque};
+
+pub fn span(
+    g: &impl GraphDCEL<
+        LinkVertex,
+        LinkDart,
+        LinkFace,
+        LinkGraphIter<LinkVertex>,
+        LinkGraphIter<LinkDart>,
+        LinkGraphIter<LinkFace>,
+    >,
+    v: LinkVertex,
+) -> HashMap<LinkVertex, LinkVertex> {
+    if g.get_vertexes().count() <= 1 {
+        return HashMap::new();
+    }
+    let mut queue = VecDeque::new();
+    let mut result = HashMap::new();
+    let mut visited = HashSet::new();
+    queue.push_back(v);
+
+    while !queue.is_empty() {
+        let v = queue.pop_front().unwrap();
+        /*if visited.contains(&v) {
+            continue;
+        }*/
```

Comment:
```suggestion
```

What about this commented out code? 
If it is no longer needed, it should be deleted.

Reply:
Sure!